src/funit/CMakeLists.txt

```diff
@@ -22,7 +22,13 @@ function(funit_target_link_pfunit funit_target)
     target_link_libraries (${funit_target} PUBLIC OpenMP::OpenMP_Fortran)
   endif ()
 
-  target_link_libraries (${funit_target} PUBLIC gftl gftl-shared fargparse)
+  include(build_submodule)
+  # Find fArgParse first so that gFTL and gFTL-shared are consistent
+  build_submodule(${PROJECT_SOURCE_DIR}/extern/fArgParse PROJECT FARGPARSE TARGET FARGPARSE::fargparse)
+  build_submodule(${PROJECT_SOURCE_DIR}/extern/gFTL-shared PROJECT GFTL_SHARED TARGET GFTL_SHARED::gftl-shared)
+  build_submodule(${PROJECT_SOURCE_DIR}/extern/gFTL PROJECT GFTL TARGET GFTL::gftl)
```

Comment:
These might actually be better moved outside this function, but they do work here